traitsui/testing/tester/qt4/implementation/list_editor.py

```diff
@@ -0,0 +1,121 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+from traitsui.testing.tester import locator
+
+from traitsui.qt4.list_editor import (
+    CustomEditor,
+)
+
+
+def find_by_name_in_nested_ui(wrapper, location):
+    """ Helper function for resolving from an _IndexedCustomEditor to a
+    TargetByName.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+    location : instance of locator.TargetByName
+    """
+    new_interactor = wrapper.locate(locator.NestedUI())
+    return new_interactor.find_by_name(location.name).target
```

Comment:
This could probably be pulled out into a separate module eventually like what was done in the draft branch.  I didn't yet as for this pr it felts unnecessary to create a full separate module to house just this function (but I certainly can change that if needed).

Reply:
I agree this should be pulled out because this logic is actually toolkit agnostic: It deals with the `UI` object. Furthermore the code is repeated - it is a true duplication in this case.
Size of a module is not an issue.